iothub/device/src/Transport/Mqtt/OrderedTwoPhaseWorkQueue.cs

```diff
@@ -43,19 +45,29 @@ public Task CompleteWorkAsync(IChannelHandlerContext context, TWorkId workId)
         {
             if (!_incompleteQueue.Any())
             {
-                throw new IotHubException("Nothing to complete.", isTransient: false);
+                if (Logging.IsEnabled)
+                {
+                    Logging.Error(context, $"{nameof(CompleteWorkAsync)} called but there are no items in the queue to complete.", nameof(CompleteWorkAsync));
+                }
+
+                Debug.Fail($"{nameof(CompleteWorkAsync)} called but there are no items in the queue to complete.");
+                return TaskHelpers.CompletedTask;
             }
 
             IncompleteWorkItem incompleteWorkItem = _incompleteQueue.Peek();
-            if (incompleteWorkItem.Id.Equals(workId))
+            if (incompleteWorkItem != null)
             {
```

Comment:
TryDequeue method is not available to all supported platforms, hence added this check